### ISSUE TITLE
Kotlin Native

### ISSUE THREAD

**Issue by nectar1** [pdvrieze/xmlutil#4]
What is about supporting of Kotlin Native?

**Comment by pdvrieze** [comment]
I'm open to it if someone is willing to work on it, there may need to be different versions for different platforms (libxml2 for unix-y systems, something else for apple, yet another solution for windows, don't speak of web assembly - that possibly needs a built in parser - perhaps the android parser can be ported to Kotlin). Personally I don't need native support and I have plenty of other things to look at.

**Comment by pdvrieze** [comment]
@mauriciomeirelles In principle yes, there is no external native code involved (no libraries needed). I don't have a mac/iOs device so haven't tested.

**Comment by mauriciomeirelles** [comment]
I tried to build for iOS and looks like it is saying it does not support `ios_x64`

```
No matching variant of io.github.pdvrieze.xmlutil:core:0.82.0 was found. The consumer was configured to find a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native', attribute 'org.jetbrains.kotlin.native.target' with value 'ios_x64' but:
          - Variant 'androidApiElements-published' capability io.github.pdvrieze.xmlutil:core:0.82.0 declares an API of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'androidJvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'ios_x64')
          - Variant 'androidRuntimeElements-published' capability io.github.pdvrieze.xmlutil:core:0.82.0 declares a runtime of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'androidJvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
```

**Comment by CarsonRedeye** [comment]
We might need this soon, so I'm willing to look into it, for iOS only

**Comment by mauriciomeirelles** [comment]
Ok! Actually there were more errors than this one. Tomorrow I can send it here the full log. I can also help with checking / implementing something (if necessary)! I’m just new to KMM so not sure exactly how to help 🙃

**Comment by pdvrieze** [comment]
Btw. it is not published yet. (I'm trying to publish it properly, but github actions isn't helping too much). There should be some versions available on ossrh snapshots https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/pdvrieze/xmlutil/ but at this point I'm still trying to make the build work correctly

**Comment by CarsonRedeye** [comment]
I'll try a snapshot versions out tomorrow thx!

**Comment by mauriciomeirelles** [comment]
How can I try the snapshot also? What should I add to maeven url and to the "implementation" on dependency? I'm a little bit lost

**Comment by CarsonRedeye** [comment]
In my build.gradle.kts I've added:
```
repositories {
    google()
    mavenCentral()
    maven(url = "https://s01.oss.sonatype.org/content/repositories/snapshots/") // This
}
```

and the added the dependency like so:
```
kotlin {
    sourceSets {
        val commonMain by getting {
            dependencies {
                implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.3.0")
                implementation("io.github.pdvrieze.xmlutil:core:0.83.1-SNAPSHOT")
                implementation("io.github.pdvrieze.xmlutil:serialization:0.83.1-SNAPSHOT")
            }
        }
    }
}
```

**Comment by CarsonRedeye** [comment]
A simple example is working fine for me on iosx86 simulator

**Comment by pdvrieze** [comment]
Thanks for letting me know. I'll be releasing a proper version soon.

**Comment by CarsonRedeye** [comment]
Also working well on iosArm64 on my iPad.

**Comment by mauriciomeirelles** [comment]
Just to let you know that I also tested and it is working fine! 
Can you let us know here when you release this new version? 
Thanks so much for the help @pdvrieze and @CarsonRedeye !!!

**Comment by pdvrieze** [comment]
I've just pushed/published a RC release 0.84.0-RC1. Please test and let me know of any issues. There are also many other fixes and new features in it. @mauriciomeirelles this should help you. Github actions are a beast to set up and a single run can take 15-20 minutes.
